### ISSUE TITLE
Error while trying to see a nested variable

### ISSUE THREAD

**Issue by p4nzer96** [microsoft/vscode-data-wrangler#190]
Type: <b>Bug</b>

Unable to watch a nested variable (dictionary with strings as keys and panadas dictionary as values): the extension reports the error _Failed to load the data frame: ValueError: Buffer has wrong number of dimensions (expected 1, got 3)_

Extension version: 1.1.2
VS Code version: Code 1.89.1 (dc96b837cf6bb4af9cd736aa3af08cf8279f7685, 2024-05-07T05:13:33.891Z)
OS version: Windows_NT x64 10.0.22635
Modes:
Remote OS version: Linux x64 6.5.0-27-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8300H CPU @ 2.30GHz (8 x 2304)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.85GB (17.65GB free)|
|Process Argv|--crash-reporter-id 86cdc9cd-8c6d-4553-8ace-c1a6b119cda0|
|Screen Reader|no|
|VM|40%|

|Item|Value|
|---|---|
|Remote|SSH: ws-mark-1|
|OS|Linux x64 6.5.0-27-generic|
|CPUs|Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz (16 x 800)|
|Memory (System)|62.65GB (58.61GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:31000450
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
chatpanelc:31048052
dsvsc021:30996838
0ee40948:31013168
pythoncenvpt:31049070
a69g1124:31058053
dvdeprecation:31052039
pythonprt:31056678
dwnewjupytercf:31046870
26j00206:31048877

```

</details>

**Comment by pwang347** [comment]
Hi @p4nzer96, thanks for reporting this issue! We'll provide a fix to make sure that the data is able to at least load. If you also have any suggestions for how a nested DataFrame should look and behave in the grid, please let us know!

**Comment by pwang347** [comment]
Hi there, we've merged a fix for this in pre-release 1.3.1! Please give it a try and let us know if the issue is still present. Note that for now nested DataFrames will be simply shown using text representation, but it will no longer prevent you from loading.
